File: app/src/main/java/com/akari/ppx/xp/hook/code/purity/AvatarHook.java
package com.akari.ppx.xp.hook.code.purity;

import com.akari.ppx.xp.hook.code.SuperbHook;

import java.util.List;

import de.robv.android.xposed.XC_MethodHook;

import static com.akari.ppx.common.constant.Prefs.REMOVE_AVATAR_DECORATION;
import static de.robv.android.xposed.XposedHelpers.getIntField;
import static de.robv.android.xposed.XposedHelpers.getObjectField;

public class AvatarHook extends SuperbHook {
	@Override
	protected void onHook(ClassLoader cl) {
		hookMethod(REMOVE_AVATAR_DECORATION, "com.sup.android.mi.usercenter.model.UserInfo", "getDecorationList", new XC_MethodHook() {
			@Override
			protected void afterHookedMethod(MethodHookParam param) {
				List list = (List) param.getResult();
				if (list != null) {
					for (int i = 0; i < list.size(); i++) {
						List infos = (List) getObjectField(list.get(i), "decorationInfos");
						if (getIntField(infos.get(0), "decorationType") == 2) {
							list.remove(i);
						}
					}
				}
			}
		});
	}
}

File: app/src/main/java/com/akari/ppx/xp/hook/purity/AvatarHook.kt
@file:Suppress("unused")

package com.akari.ppx.xp.hook.purity

import com.akari.ppx.utils.check
import com.akari.ppx.utils.getIntField
import com.akari.ppx.utils.getObjectFieldAs
import com.akari.ppx.utils.hookAfterMethod
import com.akari.ppx.xp.Init.cl
import com.akari.ppx.xp.hook.SwitchHook

class AvatarHook : SwitchHook("remove_avatar_decoration") {
    override fun onHook() {
        "com.sup.android.mi.usercenter.model.UserInfo".hookAfterMethod(
            cl,
            "getDecorationList"
        ) { param ->
            val list = param.result as ArrayList<*>? ?: return@hookAfterMethod
            list.indices.reversed().forEach { i ->
                list[i].getObjectFieldAs<ArrayList<*>>("decorationInfos")[0]
                    .getIntField("decorationType").check(2) {
                        list.removeAt(i)
                    }
            }
        }
    }
}